File: app/src/main/java/com/magdenbt/collectionsbenchmark/di/MainActivity/MainComponent.java
package com.magdenbt.collectionsbenchmark.di.MainActivity;



import com.magdenbt.collectionsbenchmark.UI.ViewFlow.MainActivity;

import dagger.BindsInstance;
import dagger.Subcomponent;

@Subcomponent(modules = {MainModule.class})
public interface MainComponent {

    @Subcomponent.Factory
    interface Factory{
        MainComponent create(@BindsInstance MainActivity activity);
    }

    void inject(MainActivity activity);



}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/di/mainactivity/MainComponent.kt
package com.magdenbt.collectionsbenchmark.di.mainactivity

import com.magdenbt.collectionsbenchmark.ui.mainactivity.MainActivity
import dagger.BindsInstance
import dagger.Subcomponent

@Subcomponent(modules = [MainModule::class])
interface MainComponent {

    @Subcomponent.Factory
    interface Factory {
        fun create(@BindsInstance mainActivity: MainActivity): MainComponent
    }

    fun inject(mainActivity: MainActivity)
}
